### PR TITLE
feat: heatmap tooltip enhancements and fixes

### PR DESCRIPTION
**Pull request by darnautov**
## Summary

- Add configurable names for X and Y axes labels 
- Add extra values to the tooltip info for X and Y axes, fix `formattedValue` for the cell value
- Export `HeatmapConfig` as `RecursivePartial`

### CODE REVIEW

File: src/chart_types/heatmap/layout/config/config.ts [hunk 1]
```diff
@@ -48,6 +48,7 @@ export const config: Config = {
   timeZone: 'UTC',
 
   xAxisLabel: {
+    name: '',
```
Comment (markov00) [inline src/chart_types/heatmap/layout/config/config.ts:51]: let's use a better and informative default `X Value`

File: src/chart_types/heatmap/layout/config/config.ts [hunk 2]
```diff
@@ -63,6 +64,7 @@ export const config: Config = {
     formatter: String,
   },
   yAxisLabel: {
+    name: '',
```
Comment (markov00) [inline src/chart_types/heatmap/layout/config/config.ts:67]: let's use a better and informative default `Y Value`

File: src/chart_types/heatmap/state/selectors/tooltip.ts [hunk 3]
```diff
@@ -48,9 +48,37 @@ export const getTooltipInfoSelector = createCachedSelector(
       pickedShapes
         .filter(({ visible }) => visible)
         .forEach((shape) => {
-          const xValueLabel = config.xAxisLabel.formatter(shape.datum.x);
+          // X-axis value
           tooltipInfo.values.push({
-            label: `${shape.datum.y} - ${xValueLabel}`,
+            label: config.xAxisLabel.name,
+            color: RGBtoString(shape.fill.color),
```
Comment (markov00) [inline src/chart_types/heatmap/state/selectors/tooltip.ts:54]:
```suggestion
            color: 'transparent',
```
I think X and Y values should not be connected to the cell color

File: src/chart_types/heatmap/state/selectors/tooltip.ts [hunk 4]
```diff
@@ -48,9 +48,37 @@ export const getTooltipInfoSelector = createCachedSelector(
...
+              key: spec.id,
+            },
+            value: `${shape.datum.x}`,
+            formattedValue: config.xAxisLabel.formatter(shape.datum.x),
+          });
+
+          // Y-axis value
+          tooltipInfo.values.push({
+            label: config.yAxisLabel.name,
+            color: RGBtoString(shape.fill.color),
```
Comment (markov00) [inline src/chart_types/heatmap/state/selectors/tooltip.ts:68]: Same comment as above
```suggestion
            color: 'transparent',
```

File: src/chart_types/heatmap/state/selectors/tooltip.ts [hunk 5]
```diff
@@ -48,9 +48,37 @@ export const getTooltipInfoSelector = createCachedSelector(
...
+              specId: spec.id,
+              key: spec.id,
+            },
+            value: `${shape.datum.y}`,
+            formattedValue: config.yAxisLabel.formatter(shape.datum.y),
+          });
+
+          // Cell value
+          tooltipInfo.values.push({
+            label: '',
```
Comment (markov00) [inline src/chart_types/heatmap/state/selectors/tooltip.ts:81]: Here, on XYAxis charts we use the `spec.name` or the `spec.id` as fallback
```suggestion
            label: label: spec.name ?? spec.id,
```
You can add an optional `name` parameter on the Heatmap spec to use in this case. an empty `name` can be used if the user don't want the default parameter.

File: src/chart_types/heatmap/layout/config/config.ts (hunk 1 above)
Comment (darnautov) [reply src/chart_types/heatmap/layout/config/config.ts]: Fixed in https://github.com/elastic/elastic-charts/pull/847/commits/b8081acaaaad1740022c85d4912c52d0ea80085e

File: src/chart_types/heatmap/layout/config/config.ts (hunk 2 above)
Comment (darnautov) [reply src/chart_types/heatmap/layout/config/config.ts]: Fixed in https://github.com/elastic/elastic-charts/pull/847/commits/b8081acaaaad1740022c85d4912c52d0ea80085e

File: src/chart_types/heatmap/state/selectors/tooltip.ts (hunk 3 above)
Comment (darnautov) [reply src/chart_types/heatmap/state/selectors/tooltip.ts]: Fixed in https://github.com/elastic/elastic-charts/pull/847/commits/b8081acaaaad1740022c85d4912c52d0ea80085e

File: src/chart_types/heatmap/state/selectors/tooltip.ts (hunk 4 above)
Comment (darnautov) [reply src/chart_types/heatmap/state/selectors/tooltip.ts]: Fixed in https://github.com/elastic/elastic-charts/pull/847/commits/b8081acaaaad1740022c85d4912c52d0ea80085e

File: src/chart_types/heatmap/state/selectors/tooltip.ts (hunk 5 above)
Comment (darnautov) [reply src/chart_types/heatmap/state/selectors/tooltip.ts]: Fixed in https://github.com/elastic/elastic-charts/pull/847/commits/b8081acaaaad1740022c85d4912c52d0ea80085e

**Review by markov00 — APPROVED** [review]
LGTM

**Pull request merged**